client/components/d3chart/d3-base/index.js

```diff
@@ -26,70 +26,34 @@ import './style.scss';
  * phase' (i.e. in 'componentDidMount' and 'componentDidUpdate' methods).
  */
 export default class D3Base extends Component {
-	static propTypes = {
-		className: PropTypes.string,
-		data: PropTypes.any, // required to detect changes in data
-		drawChart: PropTypes.func.isRequired,
-		getParams: PropTypes.func.isRequired,
-		type: PropTypes.string,
-	};
-
-	state = {
-		data: null,
-		params: null,
-		drawChart: null,
-		getParams: null,
-		type: null,
-	};
-
-	chartRef = createRef();
-
-	static getDerivedStateFromProps( nextProps, prevState ) {
-		let state = {};
-
-		if ( ! isEqual( nextProps.data, prevState.data ) ) {
-			state = { ...state, data: nextProps.data };
-		}
-
-		if ( ! isEqual( nextProps.drawChart, prevState.drawChart ) ) {
-			state = { ...state, drawChart: nextProps.drawChart };
-		}
-
-		if ( ! isEqual( nextProps.getParams, prevState.getParams ) ) {
-			state = { ...state, getParams: nextProps.getParams };
-		}
-
-		if ( nextProps.type !== prevState.type ) {
-			state = { ...state, type: nextProps.type };
-		}
+	constructor( props ) {
+		super( props );
 
-		if ( ! isEmpty( state ) ) {
-			return { ...state, params: null };
-		}
-
-		return null;
+		this.chartRef = createRef();
 	}
 
 	componentDidMount() {
-		window.addEventListener( 'resize', this.updateParams );
+		window.addEventListener( 'resize', this.drawChart );
 
-		this.drawChart();
+		this.drawUpdatedChart();
 	}
 
-	shouldComponentUpdate( nextProps, nextState ) {
+	shouldComponentUpdate( nextProps ) {
 		return (
-			( nextState.params !== null && ! isEqual( this.state.params, nextState.params ) ) ||
-			! isEqual( this.state.data, nextState.data ) ||
-			this.state.type !== nextState.type
+			this.props.className !== nextProps.className ||
+			! isEqual( this.props.data, nextProps.data ) ||
+			this.props.height !== nextProps.height ||
+			this.props.type !== nextProps.type ||
+			this.props.width !== nextProps.width
```

Comment:
A check to see if the `drawChart` function has changed is a good idea.

Reply:
Done!

https://github.com/woocommerce/wc-admin/pull/961/commits/b58d725771d29ddc96b30076b70cb4adcd2b5db7